### PR TITLE
 New: Adding README.md for Annotations codebase + cleanup

### PR DESCRIPTION
**Pull request by pramodsum**
- Cleaning up annotations methods and annotations initialization
- Reordering some annotations methods so they're correctly marked as private
- Simplified how BoxAnnotations is initialized. Excerpt from README included below

Initialization
--------------
```javascript
/* global BoxAnnotations */
const boxAnnotations = new BoxAnnotations();
const annotatorConf = boxAnnotations.determineAnnotator(options, disabledAnnotationTypes);

// Construct and init annotator
const annotator = new annotatorConf.CONSTRUCTOR(options);

annotator.init(scale);
```
Where `disabledAnnotationTypes` is a string of valid annotation types to disable.

Parameters & Options
-------
```javascript
const annotator = new annotatorConf.CONSTRUCTOR({
    annotator,
    apiHost,
    token,
    container,
    file: {
        id,
        file_version: {
            id
        },
        permissions
    },
    isMobile,
    hasTouch,
    locale,
    modeButtons: { // Annotation mode buttons, i.e. point, draw, etc
        point: {
            // Accessibilty message for the point annotation mode button
            title: 'Point annotation mode',
            // CSS selector for the point annotation mode button
            selector: '.bp-btn-annotate-point'
        },
    }
});

See **Enabling/Disabling Annotations and Annotation Types** below for more details on viewer specific annotation configurations.

```
| Parameter | Description |
| --- | --- |
| annotator | Viewer-specific annotator configurations object |
| apiHost | Host for Box API calls i.e. 'https://app.box.com/api' |
| fileId | Box file ID |
| token | A string auth token, see below for details on how to generate annotator tokens with appropriate scopes |
| container | DOM node or selector where Preview should be placed |
| file | File metadata object |
| file.id | String `Box_File` id |
| file.file_version.id | String `Box_File_Version` id |
| file.permissions | File permissions object, see below on how to scope permissions  |
| modeButtons | Object containing a CSS selector and accessibility message for the annotation mode button, see parameters and options example above |

| Option | Default | Description |
| --- | --- | --- |
| isMobile | false | Whether the user's browser is on a mobile device |
| hasTouch | false | Whether the mobile browser has touch enabled |
| locale | en-US | Shared link URL |

### CODE REVIEW

File: src/lib/annotations/README.md [hunk 1]
```diff
@@ -0,0 +1,486 @@
...
+
+`annotator.toggleAnnotationHandler()` toggles annotation modes on and off. When an annotation mode is on, annotation threads will be created at that location.
+
+`annotator.disableAnnotationMode(/* String */ mode, /* HTMLElement */ buttonEl)` disables the specified annotation mode.
+
+`annotator.enableAnnotationMode(/* String */ mode, /* HTMLElement */ buttonEl)` enables the specified annotation mode.
+
+Events
+------
+The annotator object exposes `addListener` and `removeListener` for binding to events. Event listeners should be bound before `showAnnotations()` is called, otherwise events can be missed.
```
Comment (jeremypress) [inline src/lib/annotations/README.md]: I would reword this to "Events can be bound to with `addListener` and removed with `removeListener`."

File: src/lib/annotations/README.md [hunk 2]
```diff
@@ -0,0 +1,486 @@
...
+        // Do something different when annotations are fetched on a multi-page image
+    } else {}
+});
+```
+
+Supported Annotation Types
+--------------------
+Point annotations are supported on both document and image formats. Highlight comment, highlight only, and draw annotations are only supported on document formats.
+
+### Annotation Thread
```
Comment (jeremypress) [inline src/lib/annotations/README.md]: restructure

File: src/lib/annotations/doc/README.md [hunk 3]
```diff
@@ -0,0 +1,104 @@
+# Document and Presentation Annotations
+
+The document and presentation viewers supports highlight comment, highlight only, draw and point annotations.
+
+## Screenshot
+
+![Screenshot of document point annotations](../../../../images/doc_point.png)
```
Comment (jeremypress) [inline src/lib/annotations/doc/README.md]: These aren't rendering for some reason

File: src/lib/annotations/doc/README.md [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+![Screenshot of document highlight annotations](../../../../images/doc_highlight.png)
+
+![Screenshot of document draw  annotations](../../../../images/doc_draw.png)
+
+## Supported File Extensions
+
+`doc, docx, odp, ods, odt, pdf, ppt, pptx`
+
+## Events
+See BoxAnnotations events.
```
Comment (jeremypress) [inline src/lib/annotations/doc/README.md]: Make this a link?

File: src/lib/annotations/doc/README.md [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+
+## Methods
+
+The following methods are available for the document viewer.
+
+| Method Name | Explanation | Method Parameters |
+| --- | --- | --- |
+| getAnnotatedEl | Determines the annotated element in the viewer | {HTMLElement} containerEl ||
+| getLocationFromEvent | Returns an annotation location on a document from the DOM event or null if no correct annotation location can be inferred from the event. For point annotations, we return the (x, y) coordinates and page the point is on in PDF units with the lower left corner of the document as the origin. For highlight annotations, we return the PDF quad points as defined by the PDF spec and page the highlight is on | {Event} DOM event, {String} annotation type ||
```
Comment (jeremypress) [inline src/lib/annotations/doc/README.md]: Is this a method we expect people to call? Wouldn't they get the location from a create/delete event by default?

File: src/lib/annotations/doc/README.md [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+| createAnnotationThread | Creates the proper type of thread, adds it to in-memory map, and returns it. | {Annotation[]} annotations, {Object} location, {String} [annotation type] ||
+| renderAnnotationsOnPage | Override to factor in highlight types being filtered out, if disabled. Also scales annotation canvases. | {number} page number |
```
Comment (jeremypress) [inline src/lib/annotations/doc/README.md]: Not really sure what this means

File: src/lib/annotations/doc/README.md [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+| scaleAnnotationCanvases | Scales all annotation canvases for a specified page. | {number} page number||
+
+Supported Annotation Types
+--------------------
+Point, highlight comment, highlight only, and draw annotations are supported on document formats.
```
Comment (jeremypress) [inline src/lib/annotations/doc/README.md]: This is repeated from the top

File: src/lib/annotations/doc/README.md [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+
+**Methods**
+
+The following methods are available for only highlight annotation threads.
+
+| Method Name | Explanation | Method Parameters |
+| --- | --- | --- |
+| cancelFirstComment | Cancels the first comment in the thread |  ||
+| onMousedown | Mousedown handler for thread. Deletes this thread if it is still pending |  ||
+| onClick | Click handler for thread. If click is inside this highlight, set the state to be active, and return true. If not, hide the delete highlight button, set state to inactive, and reset. The 'consumed' param tracks whether or not some other click handler activated a highlight. If not, normal behavior occurs. If true, don't set the highlight to active when normally it should be activated. We don't draw active highlights in this method since we want to delay that drawing until all inactive threads have been reset | {Event} mouse event, {boolean} whether event previously activated another highlight ||
```
Comment (jeremypress) [inline src/lib/annotations/doc/README.md]: Probably remove since these all lead to events

File: src/lib/annotations/doc/README.md [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+
+### Annotation Dialog
+
+**Methods**
+
+The following methods are available for only highlight annotation dialogs.
+
+| Method Name | Explanation | Method Parameters |
+| --- | --- | --- |
+| hideCommentsDialog | Set the state of the dialog so comments are hidden, if they're currently shown |  ||
```
Comment (jeremypress) [inline src/lib/annotations/doc/README.md]: which dialog are these referring to? All?

File: src/lib/annotations/doc/README.md [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+| drawAnnotation | Emit the message to create a highlight and render it |  ||
```
Comment (jeremypress) [inline src/lib/annotations/doc/README.md]: to create a drawing

File: src/lib/annotations/doc/README.md [hunk 11]
```diff
@@ -0,0 +1,104 @@
...
+| toggleHighlightDialogs | Toggles between the highlight annotations buttons dialog and the highlight comments dialog. Dialogs are toggled based on whether the highlight annotation has text comments or not |  ||
+| toggleHighlightCommentsReply | Toggles between the "Add a comment here" and "Reply" text areas in the comments dialog. This accounts for when a blank highlight is created and then the user tries to add a comment after the fact. | {boolean} Whether or not the dialog has comments ||
```
Comment (jeremypress) [inline src/lib/annotations/doc/README.md]: Don't thing anyone will use this

File: src/lib/annotations/doc/README.md [hunk 12]
```diff
@@ -0,0 +1,104 @@
...
+
+## Draw Annotations
+
+### Annotation Thread
+
+**Methods**
+
+The following methods are available for the annotation threads.
+
+| Method Name | Explanation | Method Parameters |
```
Comment (jeremypress) [inline src/lib/annotations/doc/README.md]: Might want to remove these for now

File: src/lib/annotations/image/README.md [hunk 13]
```diff
@@ -0,0 +1,33 @@
+# Image Annotations
+
+The image and multi-page image viewers support point annotations.
+
+## Screenshot
+
+![Screenshot of image point annotations](../../../../images/image_point.png)
```
Comment (jeremypress) [inline src/lib/annotations/image/README.md]: Screenshot isn't rendering

File: src/lib/annotations/image/README.md [hunk 14]
```diff
@@ -0,0 +1,33 @@
...
+See BoxAnnotations events.
+
+## Methods
+
+The following methods are available for the document viewer.
+
+| Method Name | Explanation | Method Parameters |
+| --- | --- | --- |
+| getAnnotatedEl | Determines the annotated element in the viewer | {HTMLElement} containerEl ||
+| getLocationFromEvent | Returns an annotation location on an image from the DOM event or null if no correct annotation location can be inferred from the event. For point annotations, we return the (x, y) coordinates for the point with the top left corner of the image as the origin. | {Event} DOM event ||
```
Comment (jeremypress) [inline src/lib/annotations/image/README.md]: Same from the doc readme comment

File: src/lib/annotations/README.md [hunk 15]
```diff
@@ -0,0 +1,486 @@
+[Box Annotations](https://developer.box.com/docs/getting-started-with-new-box-view#section-annotations)
+====================================================================
+Box Annotations allows developers to provide collaboration capabilities right from within the embedded Box preview in their application. Annotations fits a wide range of use cases and can be used to draw the reader's attention and/or provide feedback on specific parts of a document or images. To learn more about the Content Preview and for further documentation on how to use it, please go to our page on [Box Content Preview](https://developer.box.com/docs/box-content-preview).
```
[the remark this answers is not in the document]
Comment (pramodsum) [reply src/lib/annotations/README.md]: 👍

File: src/lib/annotations/image/README.md [hunk 16]
```diff
@@ -0,0 +1,33 @@
...
+| createAnnotationThread | Creates the proper type of thread, adds it to in-memory map, and returns it. | {Annotation[]} annotations, {Object} location, {String} [annotation type] ||
+| hideAllAnnotations | Hides all annotations on the image. Also hides button in header that enables point annotation mode |  |
+| showAllAnnotations | Shows all annotations on the image. Shows button in header that enables point annotation mode |  ||
+
+Supported Annotation Types
+--------------------
+Point annotations are supported on image and multi-page image formats.
```
Comment (jeremypress) [inline src/lib/annotations/image/README.md]: Repeated from the top section

File: src/lib/annotations/README.md [hunk 17]
```diff
@@ -0,0 +1,486 @@
...
+
+Content Preview supports four annotation types today - highlight comment, highlight only, draw and point annotation. Annotations are today supported on documents and image previews only. You can find the full list of supported file types for Box Content Preview at https://community.box.com/t5/Managing-Your-Content/What-file-types-and-fonts-are-supported-by-Box-s-Content-Preview/ta-p/327#FileTypesSupported.
```
[the remark this answers is not in the document]
Comment (pramodsum) [reply src/lib/annotations/README.md]: 👍

File: src/lib/annotations/README.md [hunk 18]
```diff
@@ -0,0 +1,486 @@
...
+/* global BoxAnnotations */
+const boxAnnotations = new BoxAnnotations();
+const annotatorConf = boxAnnotations.determineAnnotator(options, disabledAnnotationTypes);
+
+// Construct and init annotator
+const annotator = new annotatorConf.CONSTRUCTOR(options);
+
+annotator.init(scale);
+```
+Where `disabledAnnotationTypes` is a string of valid annotation types to disable.
```
[the remark this answers is not in the document]
Comment (pramodsum) [reply src/lib/annotations/README.md]: 👍

File: src/lib/annotations/README.md [hunk 19]
```diff
@@ -0,0 +1,486 @@
...
+Annotations can be selectively disabled through preview options. Viewer options override global showAnnotations value, for that viewer. See [Box Content Preview](https://github.com/box/box-content-preview) for more details on how to set up the Preview instances that are used with Box Annotations here.
+```
+// Turn on/off annotations for all viewers
+preview.show(..., {
+    showAnnotations: Boolean
+});
+```
+Combined with the following:
+```
+// [LEGACY] BOOLEAN - turn on/off annotations for a viewer
```
[the remark this answers is not in the document]
Comment (pramodsum) [reply src/lib/annotations/README.md]: sounds good to me!

File: src/lib/annotations/README.md (hunk 2 above)
Comment (pramodsum) [reply src/lib/annotations/README.md]: 👍

File: src/lib/annotations/doc/README.md (hunk 3 above)
Comment (pramodsum) [reply src/lib/annotations/doc/README.md]: Whoops forgot to add them... I'll add that part afterwards

File: src/lib/annotations/image/README.md (hunk 13 above)
Comment (pramodsum) [reply src/lib/annotations/image/README.md]: will add this in later as well

File: src/lib/annotations/README.md [hunk 20]
```diff
@@ -0,0 +1,474 @@
...
+====================================================================
+Box Annotations allow developers to provide collaboration capabilities right from within the embedded Box preview in their application. Annotations fits a wide range of use cases and can be used to draw the previewer's attention and/or provide feedback on specific parts of a document or images. To learn more about Box Content Preview and for further documentation on how to use it, please go to our page on [Box Content Preview](https://developer.box.com/docs/box-content-preview).
+
+Box Content Preview currently supports four annotation types - highlight comment, highlight only, draw, and point annotation. Annotations are today supported on documents and image previews only. You can find the full list of supported file types for Box Content Preview at https://community.box.com/t5/Managing-Your-Content/What-file-types-and-fonts-are-supported-by-Box-s-Content-Preview/ta-p/327#FileTypesSupported.
+
+Browser Support
+---------------
+* Desktop Chrome, Firefox, Safari, Edge, and Internet Explorer 11
+* Limited support for mobile web - previews will render but some controls may not work
```
Comment (jeremypress) [inline src/lib/annotations/README.md]: We support iOS Safari and Android Chrome

File: src/lib/annotations/README.md [hunk 21]
```diff
@@ -0,0 +1,474 @@
+[Box Annotations](https://developer.box.com/docs/getting-started-with-new-box-view#section-annotations)
+====================================================================
+Box Annotations allow developers to provide collaboration capabilities right from within the embedded Box preview in their application. Annotations fits a wide range of use cases and can be used to draw the previewer's attention and/or provide feedback on specific parts of a document or images. To learn more about Box Content Preview and for further documentation on how to use it, please go to our page on [Box Content Preview](https://developer.box.com/docs/box-content-preview).
```
Comment (jeremypress) [inline src/lib/annotations/README.md]: should change fits -> fit to be consistent
